src/ray/raylet/lineage_cache.h

```diff
@@ -1,83 +1,216 @@
 #ifndef RAY_RAYLET_LINEAGE_CACHE_H
 #define RAY_RAYLET_LINEAGE_CACHE_H
 
+#include <boost/optional.hpp>
```

Comment:
cc @concretevitamin is there a good alternative for now? Assuming that we don't want to bring in abseil just yet..

Reply:
Of course.  I'd rewrite
```
+  boost::optional<const LineageEntry &> GetEntry(const TaskID &entry_id) const;
```
into 
```
// If not found, "entry" is left as nullptr.
// Typical usage:
//    LineageEntry* entry;
//    GetEntry(entry_id, &entry);
//    if (entry != nullptr) Use(entry);
void GetEntry(const TaskID &entry_id, LineageEntry** entry) const;
```